lib/active_merchant/billing/gateways/sum_up.rb

```diff
@@ -0,0 +1,192 @@
+      end
+
+      def add_customer_data(post, payment, options)
+        post[:customer_id]            = options[:customer_id]
+        personal_details              = post[:personal_details] ||= {}
+        personal_details[:email]      = options[:email]
+        personal_details[:first_name] = payment&.first_name
+        personal_details[:last_name]  = payment&.last_name
+        personal_details[:tax_id]     = options[:tax_id]
+      end
+
+      def add_merchant_data(post, options)
+        # Required field: pay_to_email
+        # Description: Email address of the merchant (in this case Spreedly)
+        # to whom the payment is made.
+        post[:pay_to_email] = @options[:pay_to_email]
+        post[:redirect_url] = options[:redirect_url]
+        post[:return_url]   = options[:return_url]
+      end
+
+      def add_address(post, creditcard, options)
+        personal_details      = post[:personal_details] ||= {}
+        address               = personal_details[:address] = {}
+        address[:city]        = options[:billing_address][:city]
+        address[:state]       = options[:billing_address][:state]
+        address[:country]     = options[:billing_address][:country]
+        address[:line_1]      = options[:billing_address][:address1]
+        address[:postal_code] = options[:billing_address][:zip]
+      end
+
+      def add_invoice(post, money, options)
+        post[:checkout_reference] = options[:checkout_reference]
+        post[:amount]             = amount(money)
+        post[:currency]           = (options[:currency] || currency(money))
+        post[:description]        = options[:description]
+      end
+
+      def add_payment(post, payment, options)
+        post[:payment_type] = options[:payment_type] if PAYMENT_TYPES.include?(options[:payment_type])
+        card                = post[:card] = {}
+        card[:name]         = [payment.first_name, payment.last_name].join(' ')
+        card[:number]       = payment.number
+        card[:expiry_month] = format(payment.month, :two_digits)
+        card[:expiry_year]  = payment.year
+        card[:cvv]          = payment.verification_value
+      end
+
+      def commit(action, post, method = :post)
+        response = trigger_request(action, post.compact, method).symbolize_keys
+
+        Response.new(
+          success_from(response),
+          message_from(response),
+          response,
+          authorization: authorization_from(response),
+          test: test?,
+          error_code: error_code_from(response)
+        )
+      end
+
+      def trigger_request(action, post, method)
+        begin
+          response = parse(ssl_request(method, live_url + action, post.to_json, auth_headers))
+        rescue ResponseError => e
+          response = JSON.parse(e.response.body)
+        end
+
+        # Multiple invalid parameters
+        response = format_multiple_errors(response) if response.is_a?(Array)
```

Comment:
❓ Question

`response.is_a?(Array)` is the best way to check if the response was successful or not ?

Reply:
You're right, I'm going to change this behavior.